### ISSUE TITLE
homestead-provisioing's Nginx proxy isn't always configured

### ISSUE THREAD

**Issue by dtheobald** [Metaswitch/crest#224]
I believe that there’s some kind of install/config ordering problem between Nginx and homestead-prov whereby the Nginx config happens before homestead-prov is install and as such never installs the homestead-prov proxy. I haven't had time to investigate further, so I made this kludgy change to create-homestead-prov-nginx-config:
```
if [ -z $hs_provisioning_hostname ]
 then
-  echo "hs_provisioning_hostname is not defined"
-  exit
+  read -r hs_provisioning_domain hs_provisioning_port <<< "$(split_domain_and_port $hs_hostname)"
+  let "hs_provisioning_port=$hs_provisioning_port + 1"
+  echo "hs_provisioning_hostname is not defined. Defaulting to $hs_provisioning_domain:$hs_provisioning_port"
 fi

```

**Comment by eleanor-merry** [comment]
It's valid now for the hs_provisioning_hostname to not be set in this case (i.e. we've installed homestead-prov before setting up shared_config). 

We shouldn't fail the install therefore, but we should stop homestead-prov from running until this script has successfully started. 

The homer/homestead/homestead-prov/memento/ralf/ellis/restund scripts all need changing in this way

**Comment by eleanor-merry** [comment]
Fixed by various PRs linked in #229
